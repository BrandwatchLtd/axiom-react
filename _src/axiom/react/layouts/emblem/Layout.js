import React, { Component } from 'react';
import classnames from 'classnames';
import { enhance, addPropTypes, addClassName } from '../../utils/components';

export class Layout extends Component {
  static propTypes = {
    children: { node: true, isRequired: true },
  };

  render() {
    const { children, className } = this.props;
    const classes = classnames(className, 'ax-layout--emblem');

    return (
      <div className={ classes }>
        { children }
      </div>
    );
  }
}

export default enhance(Layout)(
  addPropTypes('global'),
  addClassName('global'),
);
