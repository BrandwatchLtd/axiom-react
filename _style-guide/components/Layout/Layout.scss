@import '~bw-axiom/design-patterns/layout/mixins';
@import '~bw-axiom/design-patterns/layout/_vars.js';
@import './vars';

.dm-layout {
  display: flex;
  flex: 1;
  flex-direction: column;
}

.dm-layout--open {
  @include screen-less-than($layout-change-point) {
    overflow-x: hidden;
  }
}
