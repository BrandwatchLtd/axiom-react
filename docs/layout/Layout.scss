@import '~bw-axiom/design-patterns/layout/_vars.js';
@import './vars';

.dm-layout {
  display: flex;
  flex: 1;
  flex-direction: column;
  background: green;
  @media screen and (max-width: $layout-change-point) {
    background: $color;
  }
}
