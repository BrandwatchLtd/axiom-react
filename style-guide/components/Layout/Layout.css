@import '../../../src/materials/breakpoints';

:root {
  --sg-header-height-sm: calc(var(--spacing-grid-size) * 12);
  --sg-header-height-lg: calc(var(--spacing-grid-size) * 25);

  --sg-sidebar-width: 15rem;
  --sg-sidebar-border-width: 0.0625rem;

  --sg-z-index-layout-header: 2;
}

.dm-layout {
  display: flex;
  flex: 1 1 auto;
  flex-direction: column;
}

.dm-layout__header {
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  flex-direction: column;
  justify-content: center;
  width: var(--sg-sidebar-width);
  height: var(--sg-header-height-lg);
  padding: 0 var(--page-padding-horizontal);
  border-right: var(--sg-sidebar-border-width) solid var(--color-context-border);
  border-bottom: var(--sg-sidebar-border-width) solid transparent;
  background-color: var(--color-context-background);
  text-align: center;
  z-index: var(--sg-z-index-layout-header);
}

.dm-layout__sidebar {
  display: flex;
  position: fixed;
  top: var(--sg-header-height-lg);
  bottom: 0;
  left: 0;
  flex-direction: column;
  width: var(--sg-sidebar-width);
  border-right: var(--sg-sidebar-border-width) solid var(--color-context-border);
  background-color: var(--color-context-background);
  z-index: var(--sg-z-index-layout-header);
}

.dm-layout__sidebar-body {
  flex: 1 1 0%;
  overflow: auto;
}

.dm-layout__sidebar-footer {
  flex: 0 1 auto;
  padding: var(--page-padding-vertical) var(--page-padding-horizontal);
}

.dm-layout__main {
  display: flex;
  position: relative;
  flex: 1 1 0%;
  flex-direction: column;
  margin-left: var(--sg-sidebar-width);
}

.dm-layout__content {
  position: relative;
  width: 100%;
  max-width: 100%;
  margin: 0 auto;
}

@media (--screen-more-than-small) {
  .dm-layout__content {
    width: calc(var(--layout-width-small) - var(--sg-sidebar-width));
  }
}

@media (--screen-more-than-medium) {
  .dm-layout__content {
    width: calc(var(--layout-width-medium) - var(--sg-sidebar-width));
  }
}

@media (--screen-more-than-large) {
  .dm-layout__content {
    width: calc(var(--layout-width-large) - var(--sg-sidebar-width));
  }
}

@media (--screen-less-than-small) {
  .dm-layout__header {
    width: 100%;
    height: var(--sg-header-height-sm);
    border-right: 0;
    border-bottom-color: var(--color-context-border);
    text-align: left;
  }

  .dm-layout__sidebar {
    top: var(--sg-header-height-sm);
    margin-left: 0;
    transform: translateX(-100%);
    transition-property: transform;
    transition-duration: var(--transition-time-slow);
    transition-timing-function: var(--transition-function);
  }

  .dm-layout__main {
    margin-top: var(--sg-header-height-sm);
    margin-left: 0;
  }

  .dm-layout--open {
    overflow-x: hidden;

    & .dm-layout__sidebar  {
      transform: translateX(-0%);
    }
  }
}
