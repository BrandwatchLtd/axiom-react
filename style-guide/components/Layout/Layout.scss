@import '~bw-axiom/design-patterns/layout/mixins';
@import '~bw-axiom/design-patterns/layout/vars';
@import './vars';

.dm-layout {
  display: flex;
  flex: 1 1 auto;
  flex-direction: column;
}

.dm-layout--open {
  @include screen-less-than($layout-change-point) {
    overflow-x: hidden;
  }
}
