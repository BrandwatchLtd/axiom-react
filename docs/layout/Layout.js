import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import constants from '../../style-guide/_helper/constants';

if (constants.__INCLUDE_CSS__) {
  require('./Layout.scss');
}

export class Layout extends Component {
  render() {
    const { children } = this.props;
    const classes = classnames('dm-layout');

    return (
      <div className={ classes }>
        { children }
      </div>
    );
  }
}

Layout.propTypes = {
  children: PropTypes.node,
};

export default Layout;
