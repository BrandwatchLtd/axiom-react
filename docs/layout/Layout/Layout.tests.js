import React from 'react';
import renderer from 'react-test-renderer';
import Layout from './';

describe('Layout', () => {
  let component;

  describe('default', () => {
    beforeEach(() => {
      component = renderer.create(
        <Layout>
          <div>Test</div>
        </Layout>).toJSON();
    });

    it('renders', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
