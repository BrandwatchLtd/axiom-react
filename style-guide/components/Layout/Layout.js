import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import './Layout.css';

export default class Layout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  static childContextTypes = {
    showSidebar: PropTypes.func,
    hideSidebar: PropTypes.func,
    toggleSidebar: PropTypes.func,
  };

  getChildContext() {
    return {
      showSidebar: () => this.showSidebar(),
      hideSidebar: () => this.hideSidebar(),
      toggleSidebar: () => this.toggleSidebar(),
    };
  }

  componentWillMount() {
    this.hideSidebar();
  }

  showSidebar() {
    this.setState({ open: true });
  }

  hideSidebar() {
    this.setState({ open: false });
  }

  toggleSidebar() {
    this.setState({ open: !this.state.open });
  }

  render() {
    const { children } = this.props;
    const { open } = this.state;
    const classes = classnames('dm-layout', {
      'dm-layout--open': open,
    });

    return (
      <div className={ classes }>
        { children }
      </div>
    );
  }
}
