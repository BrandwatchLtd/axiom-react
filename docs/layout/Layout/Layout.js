import React, { Component, PropTypes } from 'react';

if (__INCLUDE_CSS__) {
  require('./layout.scss');
}

export class Layout extends Component {
  render() {
    const { children } = this.props;

    return (
      <div className="ax-layout">
        { children }
      </div>
    );
  }
}

Layout.propTypes = {
  children: PropTypes.node,
};

export default Layout;
