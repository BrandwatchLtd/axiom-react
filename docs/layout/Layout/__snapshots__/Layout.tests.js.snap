exports[`Layout default renders 1`] = `
<div
  className="ax-layout">
  <div>
    Test
  </div>
</div>
`;
